### ISSUE TITLE
[a11y]: Drop Down Label, Open Menu announcement

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16260]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

1.55.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS, Voice Over

### Description

JAWS is not announcing the options inside the Dropdown 

After activating the dropdown, with arrow key navigation JAWS is not announcing the options from the list, Labels of the  Dropdown options are not conveyed.

### Recommended Fix
Ensure JAWS should announce the options from the dropdown

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-dropdown--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
